growthbook.py

```diff
@@ -1196,14 +1505,124 @@ def _getExperimentResult(
         if experiment.meta:
             meta = experiment.meta[variationId]
 
+        (hashAttribute, hashValue) = self._getOrigHashValue(experiment.hashAttribute, experiment.fallbackAttribute)
+
         return Result(
             featureId=featureId,
             inExperiment=inExperiment,
             variationId=variationId,
             value=experiment.variations[variationId],
             hashUsed=hashUsed,
             hashAttribute=hashAttribute,
-            hashValue=self._getOrigHashValue(hashAttribute),
+            hashValue=hashValue,
             meta=meta,
             bucket=bucket,
+            stickyBucketUsed=stickyBucketUsed
         )
+
+    def _derive_sticky_bucket_identifier_attributes(self) -> List[str]:
+        attributes = set()
+        for key in self._features:
+            feature = self._features[key]
+            for rule in feature.rules:
+                if rule.variations:
+                    attributes.add(rule.hashAttribute or "id")
+                    if rule.fallbackAttribute:
+                        attributes.add(rule.fallbackAttribute)
+        return list(attributes)
+    
+    def _get_sticky_bucket_attributes(self) -> dict:
+        attributes: Dict[str, str] = {}
+        if self._using_derived_sticky_bucket_attributes:
+            self.sticky_bucket_identifier_attributes = self._derive_sticky_bucket_identifier_attributes()
+        
+        if not self.sticky_bucket_identifier_attributes:
+            return attributes
+
+        for attr in self.sticky_bucket_identifier_attributes:
+            (_, hash_value) = self._getHashValue(attr)
+            if hash_value:
+                attributes[attr] = hash_value
+        return attributes
+
+    def _get_sticky_bucket_assignments(self) -> Dict[str, dict]:
```

Comment:
you may need to change this:
- pass in the experiment's `hashAttribute` and `fallbackAttribute`
- filter the docs to only include docs matching the user's `${hashAttribute}||${toString(hashValue)}` and `${fallbackAttribute}||${toString(fallbackValue)}`
- merge in the order of 1. fallbackAttribute docs, 2. primary hashAttribute docs so that primary overwrites fallback in case of conflict